docs/_templates/components_requirements.tmpl

```diff
@@ -1,39 +1,108 @@
 .. -*- mode: rst -*-
 
-Inmanta core and extensions versions
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The orchestrator is constituted of several components (python packages and npm packages).
-The python packages are of two types: one core package and several extension packages.
 
-{% if data['component_requirements']%}
 
-The following table shows the ranges of the Inmanta Python components, this product version is compatible with.
-The lower bound of these version constraints will remain the same for the whole lifetime of this major version. Upper bounds may be added when a new major version is released.
+{% if data['module_compatibility_ranges']%}
 
-This table is only meant as a reference for developers of Inmanta extensions and is meant to be used as input to a
-dependency resolver e.g. pip to ensure the resulting versions are compatible.
-This information is not relevant when installing the orchestrator via RPM as is recommended
-in the :ref:`install-server` section.
 
-.. list-table:: Inmanta core and extensions versions
-   :header-rows: 1
+.. only:: iso
 
-   * - .. centered:: Component
-     - .. centered:: Required version
+    Module compatibility ranges
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-   {% for key, value in data["component_requirements"] | dictsort %}
-   * - .. centered:: {{ key }}
-     - .. centered:: {{ value }}
-   {% endfor %}
+    We develop and distribute some core modules alongside the product. For these modules, we uphold the same
+    compatibility guarantees as for the product itself. Therefore, some versions of a module may be developed
+    for a specific version of the product (e.g. a breaking change in a module is intended only for the next major
+    iso release).
 
-.. note::
+    We achieve this by using a specific module versioning scheme, and we accompany each iso version
+    with a compatibility range for each of these core modules. This compatibility range declares which versions
+    of each module have been developed and tested for this product version. This does not mean that other versions
+    will not work, only that they may not.
 
-    This information is also available as a constraints file in `requirements.components.txt <./requirements.components.txt>`_
-    and in json format in the `compatibility.json <./compatibility.json>`_ file.
+    For stability reasons, when you depend on these modules, we encourage you to ensure that you use the appropriate
+    versions for the iso version you're using. This can be achieved one of two ways:
```

Comment:
I don't think we've ever discussed that. I am not opposed per se, and perhaps you're right that we should enforce it, but in practice other versions might work just as well.

Reply:
Went with: 
```
For stability reasons, when you depend on these modules, make sure you use the appropriate
versions for the iso version you're using. This can be achieved one of two ways:
```